tikv build fail on darwin

## Bug Report

```
[2021-12-21T17:57:18.712Z] error[E0432]: unresolved import `nix::unistd::gettid`

[2021-12-21T17:57:18.712Z]   --> components/file_system/src/thread_io.rs:15:27

[2021-12-21T17:57:18.712Z]    |

[2021-12-21T17:57:18.712Z] 15 | use nix::unistd::{getpid, gettid, Pid};

[2021-12-21T17:57:18.712Z]    |                           ^^^^^^

[2021-12-21T17:57:18.712Z]    |                           |

[2021-12-21T17:57:18.712Z]    |                           no `gettid` in `unistd`

[2021-12-21T17:57:18.713Z]    |                           help: a similar name exists in the module: `getpid`
```

<!-- Thanks for your bug report! Don't worry if you can't fill out all the sections. -->

### What version of TiKV are you using?
<!-- You can run `tikv-server --version` -->

### What operating system and CPU are you using?
<!-- If you're using Linux, you can run `cat /proc/cpuinfo` -->
master

### Steps to reproduce
<!-- If possible, provide a recipe for reproducing the error. A complete runnable program is good. -->

### What did you expect?

### What did happened?

--- a/components/file_system/src/thread_io.rs
+++ b/components/file_system/src/thread_io.rs
@@ -1,187 +1,218 @@
 // Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
 
-use std::borrow::BorrowMut;
-use std::cell::Cell;
-use std::fs::File;
-use std::io::{BufRead, BufReader};
-use std::path::PathBuf;
-use std::str::FromStr;
-use std::sync::{
-    atomic::{AtomicU64, Ordering},
-    Arc, Mutex, MutexGuard,
-};
-use std::time::Duration;
+#[cfg(target_os = "linux")]
+mod linux {
+    use std::{
+        borrow::BorrowMut,
+        cell::Cell,
+        fs::File,
+        io::{BufRead, BufReader},
+        path::PathBuf,
+        str::FromStr,
+        sync::{
+            atomic::{AtomicU64, Ordering},
+            Arc, Mutex, MutexGuard,
+        },
+        time::Duration,
+    };
 
-use nix::unistd::{getpid, gettid, Pid};
-use strum::EnumCount;
-use thread_local::ThreadLocal;
+    use nix::unistd::{getpid, gettid, Pid};
+    use strum::EnumCount;
+    use thread_local::ThreadLocal;
 
-use crate::{IOBytes, IOType};
+    use crate::{IOBytes, IOType};
 
-lazy_static! {
-    static ref THREAD_IO_SENTINEL_VEC: ThreadLocal<Arc<ThreadIOSentinel>> = ThreadLocal::new();
-    static ref THREAD_IO_TOTAL: [AtomicIOBytes; IOType::COUNT] = Default::default();
-}
-
-thread_local! {
-    static THREAD_IO_SENTINEL: Arc<ThreadIOSentinel> = THREAD_IO_SENTINEL_VEC.get_or(||
-        Arc::new(ThreadIOSentinel::current_default())
-    ).clone();
-
-    static THREAD_IO_PRIVATE: Cell<ThreadIOPrivate> = Cell::new(ThreadIOPrivate::default())
-}
-
-struct ThreadIOSentinel {
-    id: ThreadID,
-    bytes: [AtomicIOBytes; IOType::COUNT],
-}
-
-impl ThreadIOSentinel {
-    fn current_default() -> ThreadIOSentinel {
-        ThreadIOSentinel {
-            id: ThreadID::current(),
-            bytes: Default::default(),
-        }
-    }
-}
-
-struct ThreadID {
-    pid: Pid,
-    tid: Pid,
-}
-
-impl ThreadID {
-    fn current() -> ThreadID {
-        ThreadID {
-            pid: getpid(),
-            tid: gettid(),
-        }
-    }
-}
-
-#[derive(Clone, Copy)]
-struct ThreadIOPrivate {
-    io_type: IOType,
-    last_bytes: IOBytes,
-}
-
-struct AtomicIOBytes {
-    read: AtomicU64,
-    write: AtomicU64,
-}
-
-impl Default for AtomicIOBytes {
-    fn default() -> Self {
-        let bytes = IOBytes::default();
-
-        AtomicIOBytes {
-            read: AtomicU64::new(bytes.read),
-            write: AtomicU64::new(bytes.write),
-        }
-    }
-}
-
-impl AtomicIOBytes {
-    fn load(&self, order: Ordering) -> IOBytes {
-        IOBytes {
-            read: self.read.load(order),
-            write: self.write.load(order),
-        }
+    lazy_static! {
+        static ref THREAD_IO_SENTINEL_VEC: ThreadLocal<Arc<ThreadIOSentinel>> = ThreadLocal::new();
+        static ref THREAD_IO_TOTAL: [AtomicIOBytes; IOType::COUNT] = Default::default();
     }
 
-    fn store(&self, val: IOBytes, order: Ordering) {
-        self.read.store(val.read, order);
-        self.write.store(val.write, order);
+    thread_local! {
+        static THREAD_IO_SENTINEL: Arc<ThreadIOSentinel> = THREAD_IO_SENTINEL_VEC.get_or(||
+            Arc::new(ThreadIOSentinel::current_default())
+        ).clone();
+
+        static THREAD_IO_PRIVATE: Cell<ThreadIOPrivate> = Cell::new(ThreadIOPrivate::default())
     }
 
-    fn fetch_add(&self, other: IOBytes, order: Ordering) {
-        self.read.fetch_add(other.read, order);
-        self.write.fetch_add(other.write, order);
+    pub fn fetch_all_thread_io_bytes(io_type: IOType) -> IOBytes {
+        THREAD_IO_SENTINEL_VEC.iter().for_each(|sentinel| {
+            flush_sentinel(sentinel, None);
+        });
+        THREAD_IO_TOTAL[io_type as usize].load(Ordering::Relaxed)
     }
-}
 
-pub fn fetch_all_thread_io_bytes(io_type: IOType) -> IOBytes {
-    THREAD_IO_SENTINEL_VEC.iter().for_each(|sentinel| {
-        flush_sentinel(sentinel, None);
-    });
-    THREAD_IO_TOTAL[io_type as usize].load(Ordering::Relaxed)
-}
+    pub fn set_io_type(new_io_type: IOType) {
+        THREAD_IO_SENTINEL.with(|sentinel| {
+            flush_sentinel(sentinel, Some(new_io_type));
+        })
+    }
 
-fn flush_sentinel(sentinel: &ThreadIOSentinel, new_io_type: Option<IOType>) {
-    let io_bytes = fetch_exact_thread_io_bytes(&sentinel.id);
+    pub fn get_io_type() -> IOType {
+        THREAD_IO_PRIVATE.with(|private| private.get().io_type)
+    }
 
-    THREAD_IO_PRIVATE.with(|private_cell| {
-        let mut private = private_cell.get();
-        let io_type = private.io_type;
-        let last_io_bytes = private.last_bytes;
+    struct ThreadIOSentinel {
+        id: ThreadID,
+        bytes: [AtomicIOBytes; IOType::COUNT],
+    }
 
-        sentinel.bytes[io_type as usize].fetch_add(io_bytes - last_io_bytes, Ordering::Relaxed);
-        THREAD_IO_TOTAL[io_type as usize].fetch_add(io_bytes - last_io_bytes, Ordering::Relaxed);
-        private.last_bytes = io_bytes;
-        if let Some(new_io_type) = new_io_type {
-            private.io_type = new_io_type;
-        }
-        private_cell.set(private)
-    })
-}
-
-pub fn set_io_type(new_io_type: IOType) {
-    THREAD_IO_SENTINEL.with(|sentinel| {
-        flush_sentinel(sentinel, Some(new_io_type));
-    })
-}
-
-pub fn get_io_type() -> IOType {
-    THREAD_IO_PRIVATE.with(|private| private.get().io_type)
-}
-
-fn fetch_thread_io_bytes(_io_type: IOType) -> IOBytes {
-    fetch_exact_thread_io_bytes(&ThreadID::current())
-}
-
-fn fetch_exact_thread_io_bytes(id: &ThreadID) -> IOBytes {
-    let io_file_path = PathBuf::from("/proc")
-        .join(format!("{}", id.pid))
-        .join("task")
-        .join(format!("{}", id.tid))
-        .join("io");
-
-    if let Ok(io_file) = File::open(io_file_path) {
-        let reader = BufReader::new(io_file);
-        let mut io_bytes = IOBytes::default();
-
-        for line in reader.lines().flatten() {
-            if line.is_empty() || !line.contains(' ') {
-                continue;
+    impl ThreadIOSentinel {
+        fn current_default() -> ThreadIOSentinel {
+            ThreadIOSentinel {
+                id: ThreadID::current(),
+                bytes: Default::default(),
             }
-            let mut s = line.split_whitespace();
+        }
+    }
 
-            if let (Some(field), Some(value)) = (s.next(), s.next()) {
-                if let Ok(value) = u64::from_str(value) {
-                    match &field[..field.len() - 1] {
-                        "read_bytes" => io_bytes.read = value,
-                        "write_bytes" => io_bytes.write = value,
-                        _ => continue,
+    struct ThreadID {
+        pid: Pid,
+        tid: Pid,
+    }
+
+    impl ThreadID {
+        fn current() -> ThreadID {
+            ThreadID {
+                pid: getpid(),
+                tid: gettid(),
+            }
+        }
+    }
+
+    #[derive(Clone, Copy)]
+    struct ThreadIOPrivate {
+        io_type: IOType,
+        last_bytes: IOBytes,
+    }
+
+    struct AtomicIOBytes {
+        read: AtomicU64,
+        write: AtomicU64,
+    }
+
+    impl Default for AtomicIOBytes {
+        fn default() -> Self {
+            let bytes = IOBytes::default();
+
+            AtomicIOBytes {
+                read: AtomicU64::new(bytes.read),
+                write: AtomicU64::new(bytes.write),
+            }
+        }
+    }
+
+    impl AtomicIOBytes {
+        fn load(&self, order: Ordering) -> IOBytes {
+            IOBytes {
+                read: self.read.load(order),
+                write: self.write.load(order),
+            }
+        }
+
+        fn store(&self, val: IOBytes, order: Ordering) {
+            self.read.store(val.read, order);
+            self.write.store(val.write, order);
+        }
+
+        fn fetch_add(&self, other: IOBytes, order: Ordering) {
+            self.read.fetch_add(other.read, order);
+            self.write.fetch_add(other.write, order);
+        }
+    }
+
+    fn flush_sentinel(sentinel: &ThreadIOSentinel, new_io_type: Option<IOType>) {
+        let io_bytes = fetch_exact_thread_io_bytes(&sentinel.id);
+
+        THREAD_IO_PRIVATE.with(|private_cell| {
+            let mut private = private_cell.get();
+            let io_type = private.io_type;
+            let last_io_bytes = private.last_bytes;
+
+            sentinel.bytes[io_type as usize].fetch_add(io_bytes - last_io_bytes, Ordering::Relaxed);
+            THREAD_IO_TOTAL[io_type as usize]
+                .fetch_add(io_bytes - last_io_bytes, Ordering::Relaxed);
+            private.last_bytes = io_bytes;
+            if let Some(new_io_type) = new_io_type {
+                private.io_type = new_io_type;
+            }
+            private_cell.set(private)
+        })
+    }
+
+    pub fn fetch_thread_io_bytes(_io_type: IOType) -> IOBytes {
+        fetch_exact_thread_io_bytes(&ThreadID::current())
+    }
+
+    fn fetch_exact_thread_io_bytes(id: &ThreadID) -> IOBytes {
+        let io_file_path = PathBuf::from("/proc")
+            .join(format!("{}", id.pid))
+            .join("task")
+            .join(format!("{}", id.tid))
+            .join("io");
+
+        if let Ok(io_file) = File::open(io_file_path) {
+            let reader = BufReader::new(io_file);
+            let mut io_bytes = IOBytes::default();
+
+            for line in reader.lines().flatten() {
+                if line.is_empty() || !line.contains(' ') {
+                    continue;
+                }
+                let mut s = line.split_whitespace();
+
+                if let (Some(field), Some(value)) = (s.next(), s.next()) {
+                    if let Ok(value) = u64::from_str(value) {
+                        match &field[..field.len() - 1] {
+                            "read_bytes" => io_bytes.read = value,
+                            "write_bytes" => io_bytes.write = value,
+                            _ => continue,
+                        }
                     }
                 }
             }
+
+            return io_bytes;
         }
 
-        return io_bytes;
+        IOBytes::default()
     }
 
-    IOBytes::default()
-}
-
-impl Default for ThreadIOPrivate {
-    fn default() -> ThreadIOPrivate {
-        ThreadIOPrivate {
-            io_type: IOType::Other,
-            last_bytes: IOBytes::default(),
+    impl Default for ThreadIOPrivate {
+        fn default() -> ThreadIOPrivate {
+            ThreadIOPrivate {
+                io_type: IOType::Other,
+                last_bytes: IOBytes::default(),
+            }
         }
     }
 }
 
+#[cfg(target_os = "linux")]
+pub use self::linux::{fetch_all_thread_io_bytes, get_io_type, set_io_type};
+
+#[cfg(target_os = "linux")]
+use self::linux::fetch_thread_io_bytes;
+
+#[cfg(not(target_os = "linux"))]
+mod notlinux {
+    use crate::{IOBytes, IOType};
+
+    pub fn fetch_all_thread_io_bytes(io_type: IOType) -> IOBytes {
+        IOBytes::default()
+    }
+
+    pub fn set_io_type(new_io_type: IOType) {}
+
+    pub fn get_io_type() -> IOType {
+        IOType::Other
+    }
+}
+
+#[cfg(not(target_os = "linux"))]
+pub use self::notlinux::{fetch_all_thread_io_bytes, get_io_type, set_io_type};
+
+#[cfg(target_os = "linux")]
 #[cfg(test)]
 mod tests {
     use libc::O_DIRECT;
@@ -192,6 +223,8 @@ mod tests {
         os::unix::prelude::OpenOptionsExt,
     };
     use tempfile::{tempdir, tempdir_in};
+
+    use crate::IOType;
 
     use super::*;
 

--- a/components/file_system/src/thread_io.rs
+++ b/components/file_system/src/thread_io.rs
@@ -195,7 +195,7 @@ pub use self::linux::{fetch_all_thread_io_bytes, get_io_type, set_io_type};
 use self::linux::fetch_thread_io_bytes;
 
 #[cfg(not(target_os = "linux"))]
-mod notlinux {
+mod non_linux {
     use crate::{IOBytes, IOType};
 
     pub fn fetch_all_thread_io_bytes(io_type: IOType) -> IOBytes {
@@ -210,7 +210,7 @@ mod notlinux {
 }
 
 #[cfg(not(target_os = "linux"))]
-pub use self::notlinux::{fetch_all_thread_io_bytes, get_io_type, set_io_type};
+pub use self::non_linux::{fetch_all_thread_io_bytes, get_io_type, set_io_type};
 
 #[cfg(target_os = "linux")]
 #[cfg(test)]

